rotkehlchen/tests/unit/decoders/test_zerox.py

```diff
@@ -1399,4 +1452,242 @@ def test_swap_settler_token_token_ethereum(ethereum_inquirer, ethereum_accounts)
+        timestamp=timestamp,
+        location=Location.BINANCE_SC,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=Asset('eip155:56/erc20:0xB11E6ed6f481fD954Dcd4585d8877aCe06acadfE'),
+        amount=FVal(received_amount),
+        location_label=user_address,
+        notes=f'Receive {received_amount} DFlow as the result of a swap via the 0x protocol',
+        counterparty=CPT_ZEROX,
+        address=settler_address,
+    )]
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('optimism_accounts', [['0xBa41b0e948d681e4dDE13e4B153d89b611e33163']])
+def test_swap_via_settler_on_optimism(optimism_inquirer, optimism_accounts):
+    """Check that a swap via settler on optimism is decoded correctly."""
+    tx_hash = deserialize_evm_tx_hash('0x8b5e0f8a9fa216fad3687ec221c2044a7f6755f299e185e5655c762ce9689ddb')   # noqa: E501
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=optimism_inquirer, tx_hash=tx_hash)
+    user_address, timestamp, gas_amount, swap_amount, received_amount, settler_address = optimism_accounts[0], TimestampMS(1742482043000), '0.000000261927407492', '0.00145', '3.213355834788683529', '0x402867B638339ad8Bec6e5373cfa95Da0b462c85'  # noqa: E501
+    assert events == [EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=0,
+        timestamp=timestamp,
+        location=Location.OPTIMISM,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_ETH,
+        amount=FVal(gas_amount),
+        location_label=user_address,
+        notes=f'Burn {gas_amount} ETH for gas',
+        counterparty=CPT_GAS,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=1,
+        timestamp=timestamp,
+        location=Location.OPTIMISM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.SPEND,
+        asset=A_ETH,
+        amount=FVal(swap_amount),
+        location_label=user_address,
+        notes=f'Swap {swap_amount} ETH via the 0x protocol',
+        counterparty=CPT_ZEROX,
+        address=settler_address,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=2,
+        timestamp=timestamp,
+        location=Location.OPTIMISM,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=A_OP,
+        amount=FVal(received_amount),
+        location_label=user_address,
+        notes=f'Receive {received_amount} OP as the result of a swap via the 0x protocol',
+        counterparty=CPT_ZEROX,
+        address=settler_address,
+    )]
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('polygon_pos_accounts', [['0xd5c6952b0d57830118Ac3F32b6b31E8288bEF9BB']])
+def test_swap_via_settler_on_polygon_pos(polygon_pos_inquirer, polygon_pos_accounts):
+    """Check that a swap via settler on optimism is decoded correctly."""
```

Comment:
the docstring is off but no need to add those docstrings, the test title is fine here and in the other cases